lib/elixir/lib/gen_server.ex

```diff
@@ -596,11 +599,11 @@ defmodule GenServer do
   This callback is optional. If one is not implemented, the server will fail
   if a continue instruction is used.
   """
-  @callback handle_continue(continue :: term, state :: term) ::
+  @callback handle_continue(continue_arg, state :: term) ::
               {:noreply, new_state}
-              | {:noreply, new_state, timeout | :hibernate | {:continue, term}}
+              | {:noreply, new_state, timeout | :hibernate | {:continue, continue_arg}}
               | {:stop, reason :: term, new_state}
-            when new_state: term
+            when new_state: term, continue_arg: term
```

Comment:
I'm in doubt whether we can use the same variable for two `continue_arg` values that could be different.

Reply:
We cannot but in this case they are the same. We can only return a continue_arg that `handle_continue` accepts.